### PR TITLE
Add preventNamespace, move defaultNamespace and mergeNamespace to separate files

### LINKED ISSUE #120: provide a clear way to dispatch a global action from a namespaced injectable
this is useful for affecting all widgets from a single widget (must be mentioned in the docs!)

`attachNamespace(null, action)` might work, but i think there should be an explicit function for this.

we should ensure that any mechanism where the namespace can be overridden is using `defaultNamespace` (notably actions dispatched via `namespacedConnect`, injected middleware, epics)

`defaultNamespace` should include tests checking that if an action has `namespace: null`, it won't be changed

is `namespace: null` sufficient? perhaps a constant would be better.

proposed names for the function:
- `unsetNamespace`
- `removeNamespace`
- `dissocNamespace`
- `detachNamespace`

the issue with these names is that the behaviour is not reflected in the name. we are not unsetting a present namespace. we are preventing a namespace to be set automatically.

maybe something like
- `globalizeAction`
- `preventNamespace`

i'm inclining towards `preventNamespace` having an explicit constant (`NAMESPACE_PREVENTED`)

### PR DESCRIPTION
**Pull request by vwerne**
Closes #120

### CODE REVIEW

File: docs/packages/namespaces.md [hunk 1]
```diff
@@ -125,3 +125,15 @@ const state = {
 
 getStateByNamespace('namespaces', 'foo', state); // { value: 'bar' }
 ```
+
+### preventNamespace()
+
+Associates an action with a default namespace, overwriting any previous namespace..
```
Comment (wafflepie) [inline docs/packages/namespaces.md]:
```suggestion
Associates an action with a "global" namespace, ensuring that this action's namespace won't be overwritten by any namespaced injectables.
```

File: docs/packages/namespaces.md [hunk 2]
```diff
@@ -125,3 +125,15 @@ const state = {
...
+
+**Arguments**
+
+1. `action` ( _Action_ ): Action to default the namespace of.
```
Comment (wafflepie) [inline docs/packages/namespaces.md]:
```suggestion
1. `action` ( _Action_ ): Action to prevent the namespace of.
```

File: package.json [hunk 3]
```diff
@@ -34,7 +34,7 @@
     "eslint-config-prettier": "6.10.0",
     "eslint-config-react-union": "0.15.2",
     "eslint-plugin-babel": "5.3.0",
-    "eslint-plugin-import": "2.20.1",
+    "eslint-plugin-import": "2.20.0",
```
Comment (wafflepie) [inline package.json]: what

File: packages/namespaces/src/constants.js [hunk 4]
```diff
@@ -1 +1,2 @@
 export const DEFAULT_FEATURE = 'namespaces';
+export const NAMESPACE_PREVENTED = 'namespacePrevented';
```
Comment (wafflepie) [inline packages/namespaces/src/constants.js]:
```suggestion
export const NAMESPACE_PREVENTED = 'NAMESPACE_PREVENTED';
```

File: packages/namespaces/src/isActionFromNamespace.js [hunk 5]
```diff
@@ -1,10 +1,12 @@
 import { curry } from 'ramda';
 
+import { NAMESPACE_PREVENTED } from './constants';
 import getNamespaceByAction from './getNamespaceByAction';
 
 /**
  * An action is from a namespace if the passed namespace is nil (it's a global reducer/epic),
- * or if the action's namespace is nil (it's a global action) or if the namespaces match.
+ * or if the action's namespace is nil (it's a global action) or if the namespaces match
+ * or if the action's namespace is namespace prevented default.
```
Comment (wafflepie) [inline packages/namespaces/src/isActionFromNamespace.js]:
```suggestion
 * or if the action's namespace has been prevented.
```

File: packages/namespaces/src/preventNamespace.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+import { NAMESPACE_PREVENTED } from './constants';
+
+/**
+ * Associates an action with a default namespace, overwriting any previous namespace.
+ *
+ * @param {Object} action action to add the default namespace to
+ */
+const preventNamespace = action =>
+	mergeDeepLeft({ meta: { namespace: NAMESPACE_PREVENTED } }, action);
```
Comment (wafflepie) [inline packages/namespaces/src/preventNamespace.js]: please use `mergeNamespace` here so we can work with functions too. just like in attachNamespace

File: packages/namespaces/src/isActionFromNamespace.test.js [hunk 7]
```diff
@@ -1,7 +1,9 @@
+import { NAMESPACE_PREVENTED } from './constants';
 import isActionFromNamespace from './isActionFromNamespace';
 
 const fooAction = { meta: { namespace: 'foo' } };
 const barAction = { meta: { namespace: 'bar' } };
+const defaultNamespaceAction = { meta: { namespace: NAMESPACE_PREVENTED } };
```
Comment (wafflepie) [inline packages/namespaces/src/isActionFromNamespace.test.js]: NAMESPACE_PREVENTED is not a "default namespace" it is more like a namespace with special treatment. rename this to `actionWithPreventedNamespace` and update test descriptions accordingly ("action had its namespace prevented")

File: docs/packages/namespaces.md (hunk 2 above)
Comment (vwerne) [reply docs/packages/namespaces.md]: done

File: packages/namespaces/src/isActionFromNamespace.test.js (hunk 7 above)
Comment (vwerne) [reply packages/namespaces/src/isActionFromNamespace.test.js]: done

File: packages/namespaces/src/preventNamespace.js (hunk 6 above)
Comment (vwerne) [reply packages/namespaces/src/preventNamespace.js]: done

File: packages/namespaces/src/isActionFromNamespace.js (hunk 5 above)
Comment (vwerne) [reply packages/namespaces/src/isActionFromNamespace.js]: done

File: packages/namespaces/src/constants.js (hunk 4 above)
Comment (vwerne) [reply packages/namespaces/src/constants.js]: done

File: package.json (hunk 3 above)
Comment (vwerne) [reply package.json]: windows issues
